server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexRequest.java

```diff
@@ -114,4 +118,22 @@ public void writeTo(StreamOutput out) throws IOException {
         indicesOptions.writeIndicesOptions(out);
         waitForActiveShards.writeTo(out);
     }
+
+    public static CloseIndexRequest fromRestRequest(RestRequest request) throws IOException {
+        String[] indices = request.hasParam("index")
+            ? Strings.splitStringByCommaToArray(request.param("index"))
```

Comment:
If a request mentions indices in both the URL and the body then we should definitely not silently ignore one of those collections. IMO we should reject requests that use both forms.

Reply:
Ok. I added a better validation, and the requests are rejected when indices are specified both in path and body.